engine/execution/computation/computer/uploader/gcp_uploader.go

```diff
@@ -45,14 +45,21 @@ func (u *GCPBucketUploader) Upload(computationResult *execution.ComputationResul
 	object := u.bucket.Object(objectName)
 
 	writer := object.NewWriter(u.ctx)
-	defer func() {
-		err := writer.Close()
-		if err != nil {
-			u.log.Warn().Err(err).Str("object_name", objectName).Msg("error while closing GCP object")
-		}
-	}()
-
-	return WriteComputationResultsTo(computationResult, writer)
+
+	// serialize and write computation result to upload stream
+	err := WriteComputationResultsTo(computationResult, writer)
+	if err != nil {
+		return fmt.Errorf("failed to write computation result to GCP stream: %w", err)
```

Comment:
if this error block is reached, is `Close()` still called? imho the `Close()` should still be called in a defer, so that in any case the writer will be closed. in that case we can set error from `Close()` to a named return value(error) of this function, or we use multi-error which @janezpodhostnik has been working on.